### ISSUE TITLE
[Design implementation] Incorrect ownership and permissions for participation key

### ISSUE THREAD

**Issue by jecassis** [algorand/go-algorand#79]
### Subject of the issue
If running with `“shared_server": true` then `goal addpartkey` generates a file with ownership `$USER:$USER` and permissions 644. Consequently node fails to restart with a 'cannot load readonly database' error.

Message in `node.log`:
```
{"file":"accountManager.go","function":"github.com/algorand/go-algorand/data.(*AccountManager).DeleteOldKeys","level":"warning","line":130,"msg":"AccountManager.DeleteOldKeys(205159): key for <address> (0-3000000): Participation.DeleteOldKeys: failed to update account: attempt to write a readonly database","time":"2019-06-22T16:58:30.314799Z"}
```

Partial output of `ls -alF`:
```
/var/lib/algorand/mainnet-v1.0:
drwxrwx--- 2 algorand algorand     4096 Jun 23 07:48 ./
drwxrwx--- 7 algorand algorand     4096 Jun 23 06:00 ../
-rw-r--r-- 1 <user>     <user>     3563520 Jun 23 07:48 <address>.0.3000000.partkey
```

### Your environment
* Software version: `4294967323 1.0.27.stable [rel/stable] (commit #ee904b7b)`
* Node status: Not applicable
* Operating System: `Linux <hostname> 4.15.0-52-generic #56-Ubuntu SMP Tue Jun 4 22:49:08 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux`

### Steps to reproduce
As user:
```
$ goal account addpartkey ...
$ goal account listpartkeys ...
$ algod -d $ALGORAND_DATA
$ less -n +F $ALGORAND_DATA/node.log
```

### Expected behaviour
Key should be added, ownership changed to `algorand` user and permissions set to 664 or 660. With 644 permissions and ownership `algorand`, `goal account listpartkeys` fails to list the key.

### Actual behaviour
See above, but in summary `algod` can't read key and fails to start.

**Comment by zeldovich** [comment]
Yeah, we should clean up the handling of partkeys.

Really, goal shouldn't be directly poking at the data directory, and we should add a REST API for managing partkeys.

In your issue report, where did the `algorand` group come from?  Is this a modified version of the algorand debian package that you've extended along the lines of what you were suggesting in #77?  Otherwise, in the default install right now, I don't think users can run `goal account addpartkey` successfully, because they can't write to `/var/lib/algorand`.

**Comment by jecassis** [comment]
Correct, we had found ourselves aliasing `sudo -u algorand goal` for _all_ the goal commands, eventually, this causes the state of the install to become unstable. Furthermore, multiple users can see other users' wallets (which is a no-no for us). One particular issue we ran into was that the `clerk` commands were not seeing an address as multisig (i.e. transactions would not contain `msig` and signing fails with version not matching 1). So I added the `algorand` group and added the users to it. This is our stopgap solution to prevent issues with other commands and multiple users stepping over each other.
